storage/src/storage.rs

```diff
@@ -22,3 +24,108 @@ pub trait Storage<ConnData: Debug, Key, Value> {
     /// Delete an entry from the storage, identified by its key.
     fn delete(&mut self, key: Key) -> StorageResult<()>;
 }
+
+/// Trait which marks a type as storable.
+/// The simplest way to implement this trait
+/// is to add `#[derive(Serialize, Deserialize)]` to the type definition.
+/// The storage works on raw bytes, so we need to serialize and deserialize the data.
+/// The default implementation uses MessagePack, but the implementor is free to chose
+/// a different encoding for their custom types.
+pub trait Storable: Sized {
+    /// Convert `Self` into `Vec<u8>`
+    fn to_bytes(&self) -> StorageResult<Vec<u8>>;
+    /// Convert `Vec<u8>` into `Self
+    fn from_bytes(x: &[u8]) -> StorageResult<Self>;
+}
+
+// By default, mark as `Storable` all the types which can be serialized and deserialized
```

Comment:
// By default, mark all the types which can be serialized and deserialized
// using serde as `Storable`.